resources/windows/wix/project.wxs.erb

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="windows-1252"?>
+  <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
+    <Product
+      Id="*"
+      UpgradeCode="<%= settings[:upgrade_code] %>"
+      Name="<%= settings[:product_name] %>"
+      Language="1033"
+      Codepage="1252"
+      Version="<%= @version.sub(/\.g[0-9a-z]{7}/, '') %>"
+      Manufacturer="<%= settings[:company_name] %>"
+    />
+    <Package
+      InstallerVersion="300"
+      InstallScope="perMachine"
+      Description="<%= "#{settings[:product_word]}#{@platform.architecture == "x64" ? " (64-bit)" : ""}" %> Installer"
+      Comments="<%= @homepage %>"
+      Compressed="yes"
+      Platform="<%= @platform.architecture %>"
+    />
+    <MajorUpgrade AllowDowngrades="yes" />
+    <Media Id="1" Cabinet="<%= settings[:product_word] %>.cab" EmbedCab="yes" CompressionLevel="high" />
+
+    <Feature Id="<%= settings[:product_word] %>Runtime" Title="<%= settings[:product_word] %> Runtime" Level="1">
+      <!-- All heat runs and WiX Fragment files containing Component elements will reference this ComponentGroup -->
+      <ComponentGroup Id="ProductComponentGroup" />
+    </Feature>
+
+    <!-- We will use DirectoryRef at the project level to hook in the project directory structure -->
+    <Directory Id='TARGETDIR' Name='SourceDir' >
```

Comment:
Is it possible that different projects are going to want to set this as a different name? So, if a project wanted `Name='sourcedir'` or `Name='builddir'` or something?

I'm thinking we could have `proj.setting(:targetdir, "SourceDir")`, which we could then use to build the directory structure up.

```
proj.setting(:company_name, "Puppet Labs")
proj.setting(:company_id, "PuppetLabs")
proj.setting(:product_name, "Puppet Agent (64-bit)")
proj.setting(:product_id, "Puppet")

proj.setting(:targetdir, "SourceDir")
proj.setting(:base_dir, File.join(proj.targetdir, "ProgramFilesFolder")
proj.setting(:install_root, File.join(proj.base_dir, proj.company_name)
proj.setting(:prefix, File.join(proj.install_root, proj.product_id)

proj.directory proj.base_dir
proj.directory proj.install_root, proj.company_id
proj.directory proj.prefix, "INSTALLDIR"
```

And then we do all the installation in a directory like `SourceDir/ProgramFilesFolder/Puppet Labs/Puppet` (ruby, openssl, facter, curl, puppet, etc).

And then @McdonaldSeanp does the directory work. `Name=<%= File.basename(directory.path) %>` and `Id=<%= directory.id %>` or something.

and we get all the directories we specify (notice I didn't include :targetdir in the list of directory declarations) which process all the subdirectories. In `project.wxs.erb`, we have this line be `<Directory Id='TARGETDIR' Name='<%= settings[:targetdir] %>' >`, and then in the directory erb we know the ID will be `TARGETDIR` and everything will be nested under it. Then things like `ruby.exe` will have a path set to  `SourceDir/ProgramFilesFolder/Puppet Labs/Puppet/bin/ruby.exe` which might make @jcoconnor's filter stuff work.

That still leaves the problem that vanagon is building packages with a space in the path name (https://github.com/puppetlabs/puppet-agent/blob/master/configs/components/openssl.rb#L81) and the possibility that some of these packages will hardcode the path into how they link, etc. But I need someone else to think about that problem for the moment.